ports/windows/windows_mphal.c

```diff
@@ -265,5 +265,15 @@ uint64_t mp_hal_time_ns(void) {
 // TODO: POSIX et al. define usleep() as guaranteedly capable only of 1s sleep:
 // "The useconds argument shall be less than one million."
 void mp_hal_delay_ms(mp_uint_t ms) {
-    usleep((ms) * 1000);
+    #ifdef MICROPY_EVENT_POLL_HOOK
+    mp_uint_t start = mp_hal_ticks_ms();
+    while (mp_hal_ticks_ms() - start < ms) {
+        // MICROPY_EVENT_POLL_HOOK does mp_hal_delay_us(500) (i.e. usleep(500)).
+        MICROPY_EVENT_POLL_HOOK
+    }
+    #else
+    // TODO: POSIX et al. define usleep() as guaranteedly capable only of 1s sleep:
```

Comment:
Should delete the comment on line 265 if this is here

Reply:
Or actually: this is Windows, not exactly known for its POSIX compliance, moreover the msvc port supplies its own implementation of usleep, so tldr: this comment was copied from the unix port but probably doesn't matter or is even incorrect here.